### PR TITLE
Fix close tabs crash ipad

### PR DESCRIPTION
**Pull request by ecotopian**
fixes https://ecosia.atlassian.net/browse/MOB-1525

### CODE REVIEW

File: Client/Frontend/Browser/Tabs/ChronologicalTabsViewController.swift [hunk 1]
```diff
@@ -147,11 +147,16 @@ extension ChronologicalTabsViewController {
...
     }
 
-    func didTapToolbarDelete(_ sender: UIBarButtonItem) {
+    func didTapToolbarDelete(_ sender: Any) {
         let controller = AlertController(title: nil, message: nil, preferredStyle: .actionSheet)
         controller.addAction(UIAlertAction(title: Strings.AppMenuCloseAllTabsTitleString, style: .default, handler: { _ in self.viewModel.closeTabsForCurrentTray() }), accessibilityIdentifier: "TabTrayController.deleteButton.closeAll")
         controller.addAction(UIAlertAction(title: Strings.CancelString, style: .cancel, handler: nil), accessibilityIdentifier: "TabTrayController.deleteButton.cancel")
-        controller.popoverPresentationController?.barButtonItem = sender
+        if let item = sender as? UIBarButtonItem {
+            controller.popoverPresentationController?.barButtonItem = item
```
Comment (ecotopian) [inline Client/Frontend/Browser/Tabs/ChronologicalTabsViewController.swift:155]: FYI: at this point sender was a `UIButton` (UIBarButtonItem with custom UIButton view) and not `UIBarButtonItem` which caused the crash 💥

File: Client/Frontend/Browser/Tabs/ChronologicalTabsViewController.swift [hunk 2]
```diff
@@ -147,11 +147,16 @@ extension ChronologicalTabsViewController {
         TelemetryWrapper.recordEvent(category: .action, method: .add, object: .tab, value: viewModel.isInPrivateMode ? .privateTab : .normalTab)
     }
 
-    func didTapToolbarDelete(_ sender: UIBarButtonItem) {
+    func didTapToolbarDelete(_ sender: Any) {
         let controller = AlertController(title: nil, message: nil, preferredStyle: .actionSheet)
         controller.addAction(UIAlertAction(title: Strings.AppMenuCloseAllTabsTitleString, style: .default, handler: { _ in self.viewModel.closeTabsForCurrentTray() }), accessibilityIdentifier: "TabTrayController.deleteButton.closeAll")
         controller.addAction(UIAlertAction(title: Strings.CancelString, style: .cancel, handler: nil), accessibilityIdentifier: "TabTrayController.deleteButton.cancel")
-        controller.popoverPresentationController?.barButtonItem = sender
```
Comment (ecotopian) [inline Client/Frontend/Browser/Tabs/ChronologicalTabsViewController.swift:154]: FYI: at this point sender was a `UIButton` (UIBarButtonItem with custom UIButton view) and not `UIBarButtonItem` which caused the crash 💥

File: Client/Frontend/Browser/GridTabViewController.swift [hunk 3]
```diff
@@ -609,15 +609,20 @@ extension GridTabViewController {
...
-    func didTapToolbarDelete(_ sender: UIBarButtonItem) {
+    func didTapToolbarDelete(_ sender: Any) {
         if tabDisplayManager.isDragging {
             return
         }
 
         let controller = AlertController(title: nil, message: nil, preferredStyle: .actionSheet)
         controller.addAction(UIAlertAction(title: Strings.AppMenuCloseAllTabsTitleString, style: .destructive, handler: { _ in self.closeTabsForCurrentTray() }), accessibilityIdentifier: "TabTrayController.deleteButton.closeAll")
         controller.addAction(UIAlertAction(title: .TabTrayCloseAllTabsPromptCancel, style: .cancel, handler: nil), accessibilityIdentifier: "TabTrayController.deleteButton.cancel")
-        controller.popoverPresentationController?.barButtonItem = sender
```
Comment (ecotopian) [inline Client/Frontend/Browser/GridTabViewController.swift:620]: FYI: at this point sender was a `UIButton` (UIBarButtonItem with custom UIButton view) and not `UIBarButtonItem` which caused the crash 💥

**Review by vauxhall — APPROVED** [review]
Nice catch, this was quite a tricky 🐛

**Review by jman82 — APPROVED** [review]
Looks awesome!! (warming up :sunglasses: ) :rocket:

**Pull request merged**